### PR TITLE
feat: Add time zone conversion to time in scratchpad

### PR DESCRIPTION
**Pull request by nowhszh**
The timestamps in the logs belong to different time zones, so add a conversion function
image

### CODE REVIEW

**Comment by nowhszh** [comment]
image
When the ScratchPad window is small, the layout becomes a bit ugly, can you give some advice, thanks! @variar

**Comment by variar** [comment]
Maybe timezone combobox could be on a separate line?

**Comment by nowhszh** [comment]
> Maybe timezone combobox could be on a separate line?


image
I've put the time zone combo box on a separate line

File: src/ui/include/scratchpad.h [hunk 1]
```diff
@@ -23,11 +23,45 @@
...
+
+  private:
+    QString displayTime();
+
+  private:
+    QFormLayout* layout_;
+    std::unique_ptr<QFormLayout> selfLayout_;
+
+    // Use the validity of QVariant to decide whether to display the time
+    std::unique_ptr<QVariant> timestamp_;
```
Comment (variar) [inline src/ui/include/scratchpad.h:59]: Is there a reason to use `unique_ptr`? And `QVariant`? Looks like it can be just `std::optional<int64>`.

**Comment by variar** [comment]
Looks good! Is it ready to merge?

**Comment by nowhszh** [comment]
> Looks good! Is it ready to merge?

Yes, it's ready.

**Pull request merged**